main/user_main.c

```diff
@@ -254,36 +303,31 @@ void ICACHE_FLASH_ATTR init_wifi(bool modeAP) {
 
 	wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
 	ESP_ERROR_CHECK( esp_wifi_init(&cfg) );
-        ESP_ERROR_CHECK( esp_wifi_set_storage(WIFI_STORAGE_RAM) );
+	ESP_ERROR_CHECK( esp_wifi_set_storage(WIFI_STORAGE_RAM) );
 
-	//Go to station mode
 	esp_wifi_disconnect();
-
-	if(modeAP) {
-		ESP_ERROR_CHECK( esp_wifi_set_mode(WIFI_MODE_AP) );
-
-		wifi_config_t ap_config;
-		strcpy((char*)(&ap_config.ap.ssid), "ESP");
-		ap_config.ap.ssid_len = 3;
-		ap_config.ap.channel = 1;
-		ap_config.ap.authmode = WIFI_AUTH_OPEN;
-		ap_config.ap.ssid_hidden = 0;
-		ap_config.ap.max_connection = 1;
-		ap_config.ap.beacon_interval = 100;
-
-		esp_wifi_set_config(WIFI_IF_AP, &ap_config);
-	}
-	else {
-		esp_wifi_set_mode(WIFI_MODE_STA);
-
-		//Connect to the defined access point.
-		wifi_config_t config;
-		memset(&config, 0, sizeof(config));
-		sprintf((char*)config.sta.ssid, "RouterSSID");			// @TODO: Changeme
-		sprintf((char*)config.sta.password, "RouterPassword"); 	// @TODO: Changeme
-		esp_wifi_set_config(WIFI_IF_STA, &config);
-		esp_wifi_connect();
-	}
+	ESP_ERROR_CHECK( esp_wifi_set_mode(mode) );
+
+	wifi_config_t ap_config;
+	strcpy((char*)(&ap_config.ap.ssid), "ESP");
+	ap_config.ap.ssid_len = 3;
+	ap_config.ap.channel = 1;
+	ap_config.ap.authmode = WIFI_AUTH_OPEN;
+	ap_config.ap.ssid_hidden = 0;
+	ap_config.ap.max_connection = 1;
+	ap_config.ap.beacon_interval = 100;
+
+	esp_wifi_set_config(WIFI_IF_AP, &ap_config);
+
+	//Connect to the defined access point.
+	wifi_config_t sta_config = {
+		.sta = {
+			.ssid = EXAMPLE_WIFI_SSID,
+			.password = EXAMPLE_WIFI_PASS
+		},
+	};
+	esp_wifi_set_config(WIFI_IF_STA, &sta_config);
+	//esp_wifi_connect();
```

Comment:
is this call to connect not necessary anymore?

Reply:
Since you just merged a bunch of WiFi related stuff, I'll revert what I did regarding WiFi.